src/printer.js

```diff
@@ -181,6 +181,18 @@ function genericPrintNoParens(path, options, print, args) {
     case "EmptyStatement":
       return "";
     case "ExpressionStatement":
+      // Detect Flow-parsed directives
+      if (n.directive) {
+        return concat([
+          nodeStr(
+            n,
+            Object.assign({}, options, {
```

Comment:
Ah, I misunderstood. Fixed in https://github.com/prettier/prettier/pull/1571/commits/a7692e9f41a4fa324a3906f07f30016356a8f97d